<a href="https://colab.research.google.com/github/jovaceos/analisis_explortorio_datos/blob/main/EDA_Trabajo_Final_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Tabla de contenido**


I.

1) Contexto

2) Pregunta SMART

3) Diccionario de datos

4) Objetivo del análisis

II.

5) Análisis exploratorio

5.1) Análisis de estructura

5.2) Definición de variables a trabajar

5.3) Análisis de variables categóricas

III.

6) Preparación de los datos

In [3]:
import math

import seaborn as sns
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LinearRegression, Ridge, Lasso, LogisticRegression #modelamiento
from sklearn.cluster import KMeans
from sklearn.mixture import GaussianMixture
from sklearn.feature_selection import SelectFromModel
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis, QuadraticDiscriminantAnalysis
from sklearn.decomposition import PCA
from IPython.display import display
from sklearn.base import BaseEstimator, TransformerMixin
from sklearn.pipeline import Pipeline
from sklearn.svm import SVC
from sklearn.metrics import roc_auc_score,roc_curve
from sklearn.preprocessing import OneHotEncoder
from sklearn.model_selection import StratifiedKFold
from sklearn.neural_network import MLPClassifier

# import warnings filter
from warnings import simplefilter
# ignore all future warnings
simplefilter(action='ignore', category=FutureWarning)
simplefilter(action='ignore', category=UserWarning)

**I**

**1) Contexto**

El modelo de medición de grupos de investigación e investigadores, tienen como propósito "*que el conocimiento generado a partir de la investigación y el desarrollo tecnológico nacional tenga un mayor impacto sobre el sistema productivo y contribuya a la solución de las problemáticas de la sociedad colombiana*". A nivel institucional, permite
generar conocimiento sobre las capacidades, fortalezas, debilidades y potencialidades de los grupos y que sirva como un instrumento para la gestión interna y para la evaluación de quienes integran el Sistema Nacional de Ciencia, Tecnología e Innovación. En relación a la tipología de productos (producción de nuevo conocimiento; productos de desarrollo tecnológico e innovación; productos de apropiación social del conocimiento y divulgación pública de la ciencia y productos de actividades relacionadas a la formación de recurso humano)

**2) Pregunta SMART**

¿Cuáles son los valores mínimos de indicadores (grupo, producción de nuevo conocimiento, desarrollo tecnológico e innovación; apropación social de conocimiento y divulgación pública de la ciencia y productos relacionados a la formación de recurso humano) que debe de cumplir un grupo de investigación para mejorar su clasificación en una próxima medición de grupos de investigación?

**Específica**: Valores mínimos que debe de cumplir un grupo de investigación para mejorar la clasificación.

**Medible**: Cantidad de productos, según tipología, que debe de cumplir el grupo de investigación para mejorar la clasificación.

**Accionable**: Identificación de productos específicos que debe de tener el grupo de investigación para mejorar la clasificación.

**Relevante**: Permite generar conocimiento sobre las capacidades, fortalezas, debilidades y potencialidades de los grupos y que sirva como un instrumento para la gestión interna y para la evaluación.

**Tiempo definido**: Próxima medición de grupos de investigación.

**3) Diccionario de datos**

Para dar respuesta a la anterior pregunta, se hará uso de dos (2) fuentes de información:

La primera, denominada, **Grupos de investigación reconocidos**, que contiene la siguiente información:

*   ID_CONVOCATORIA. Identificador de reconocimiento convocatoria. Número
*   NME_CONVOCATORIA. Nombre de convocatoria reconocimiento. Texto simple
*   ANO_CONVO. Año de inicio vigencia de convocatoria. Fecha y hora
*   COD_GRUPO_GR. Código del grupo de investigación. Texto simple
*   NME_GRUPO_GR. Nombre del grupo de investigación. Texto simple
*   FCREACION_GR	. Fecha de creación del grupo. Fecha y hora
*   NME_MUNICIPIO_GR. Nombre del municipio del grupo. Texto simple

La segunda, denominada, **Producción de grupos de investigación**, que contiene la siguiente información:

*   ID_CONVOCATORIA. Identificador de la convocatoria. Texto simple
*   NME_CONVOCATORIA. Nombre de convocatoria reconocimiento. Texto simple
*   ANO_CONVO. Año de realización de la convocatoria. Fecha y hora
*   ID_PRODUCTO_PD. Identificador único del producto. Texto simple
*   NME_CLASE_PD. Nombre de la clase de producto al que pertenece el producto. Texto simple
*   NME_TIPO_MEDICION_PD. Nombre de tipo de producto al que pertenece el producto. Texto simple
*   NME_TIPOLOGIA_PD. Nombre de la tipología obtenida del producto. Texto simple


**4) Objetivo del análisis**

Identificar los valores mínimos de indicadores (grupo, producción de nuevo conocimiento, desarrollo tecnológico e innovación; apropación social de conocimiento y divulgación pública de la ciencia y productos relacionados a la formación de recurso humano) así como la cantidad de productos mínimos que debe de cumplir un grupo de investigación para mejorar su clasificación en una próxima medición de grupos de investigación

**II**

**5)** **Análisis exploratorio**

Se procedera revisar las dos (2) fuentes de información a utilizar para dar respuesta a la pregunta SMART y así cumplir el objetivo establecido.

Primero, se procederá a cargar la información correspondiente a la **clasificación** de los grupos de investigación, según las convocatorias de Minciencas para el período 2013-2021

**5.1)** **Análisis de estructura**

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [6]:
Clasificacion_grupos = pd.read_csv ('/content/drive/MyDrive/EDA_Trabajo_Final/Grupos_Investigacion_Reconocidos.csv')
Clasificacion_grupos.shape

(30051, 21)

La información consignada en el dataframe de clasificación de grupos cuenta con 21 columnas y 30051 filas. Sin embargo, cuando se pretende acceder a la información del conjunto de datos desde la url, se aprecia que la información, no viene completa, solo trae una parte de la información

In [ ]:
clasificacion=pd.read_csv('https://www.datos.gov.co/resource/hrhc-c4wu.csv', sep=',')
clasificacion.shape

(1000, 21)

In [ ]:
Clasificacion_grupos.head()

ID_CONVOCATORIA          NME_CONVOCATORIA   ANO_CONVO COD_GRUPO_GR  \
0               16  Convocatoria 640 de 2013  31/10/2013   COL0016283   
1               16  Convocatoria 640 de 2013  31/10/2013   COL0018751   
2               16  Convocatoria 640 de 2013  31/10/2013   COL0013334   
3               16  Convocatoria 640 de 2013  31/10/2013   COL0013316   
4               16  Convocatoria 640 de 2013  31/10/2013   COL0014583   

                                        NME_GRUPO_GR FCREACION_GR  \
0                          Socialización y violencia   01/02/1993   
1                                         ECITRONICA   01/01/2002   
2  Línea de Investigación en Jóvenes y Culturas J...   01/06/1995   
3                                   Género y Cultura   01/09/1998   
4              Pavimentos y Materiales de Ingeniería   01/12/2002   

  NME_MUNICIPIO_GR NME_DEPARTAMENTO_GR     NME_REGION_GR NME_PAIS_GR  ...  \
0     Bogotá, D.C.       Bogotá, D. C.  Distrito Capital    Colombia  ...   
1     Bogotá, D.C.       Bogotá, D. C.  Distrito Capital    Colombia  ...   
2     Bogotá, D.C.       Bogotá, D. C.  Distrito Capital    Colombia  ...   
3     Bogotá, D.C.       Bogotá, D. C.  Distrito Capital    Colombia  ...   
4     Bogotá, D.C.       Bogotá, D. C.  Distrito Capital    Colombia  ...   

   ID_AREA_CON_GR NME_AREA_ESP_GR  \
0              5I     No registra   
1              2B     No registra   
2              5D     No registra   
3              5I     No registra   
4              2A     No registra   

                                        NME_AREA_GR         NME_GRAN_AREA_GR  \
0                           Otras Ciencias Sociales        Ciencias Sociales   
1  Ingenierías Eléctrica, Electrónica e Informática  Ingeniería y Tecnología   
2                                        Sociología        Ciencias Sociales   
3                           Otras Ciencias Sociales        Ciencias Sociales   
4                                  Ingeniería Civil  Ingeniería y Tecnología   

  NME_CLASIFICACION_GR ORDEN_CLAS_GR  EDAD_ANOS_GR  \
0                    C             2         21.00   
1                    C             2         12.08   
2                    D             1         18.67   
3                    C             2         15.42   
4                    D             1         11.17   

                                         INST_AVAL  \
0                              UNIVERSIDAD CENTRAL   
1  ESCUELA COLOMBIANA DE INGENIERIA JULIO GARAVITO   
2                              UNIVERSIDAD CENTRAL   
3                              UNIVERSIDAD CENTRAL   
4                 UNIVERSIDAD CATOLICA DE COLOMBIA   

                                   NME_PROG_COLC1_GR  \
0  Ciencia, Tecnología e Innovación en Ciencias H...   
1  Ciencia, Tecnología e Innovación en Tecnología...   
2  Ciencia, Tecnología e Innovación en Ciencias H...   
3  Ciencia, Tecnología e Innovación en Ciencias H...   
4  Ciencia, Tecnología e innovación en Ambiente, ...   

                                NME_PROG_COLC2_GR  
0   Ciencia, Tecnología e Innovación en Educación  
1  Desarrollo Tecnológico e Innovación Industrial  
2   Ciencia, Tecnología e Innovación en Educación  
3   Ciencia, Tecnología e Innovación en Educación  
4  Desarrollo Tecnológico e Innovación Industrial  

[5 rows x 21 columns]

In [ ]:
Clasificacion_grupos.tail()

ID_CONVOCATORIA          NME_CONVOCATORIA   ANO_CONVO COD_GRUPO_GR  \
30046               21  Convocatoria 894 de 2021  25/02/2021   COL0224491   
30047               21  Convocatoria 894 de 2021  25/02/2021   COL0224571   
30048               21  Convocatoria 894 de 2021  25/02/2021   COL0224615   
30049               21  Convocatoria 894 de 2021  25/02/2021   COL0224642   
30050               21  Convocatoria 894 de 2021  25/02/2021   COL0224713   

                    NME_GRUPO_GR FCREACION_GR NME_MUNICIPIO_GR  \
30046                      ENSUP   01/01/2016            Pasto   
30047  EMPRENDIMIENTO Y NEGOCIOS   01/01/2019    No disponible   
30048                   ANGIOINV   01/12/2019    Villavicencio   
30049                HPOResearch   01/01/2019     Bogotá, D.C.   
30050    Hematología y Oncología   01/01/2019    No disponible   

      NME_DEPARTAMENTO_GR     NME_REGION_GR    NME_PAIS_GR  ...  \
30046              Nariño          Pacífico       Colombia  ...   
30047       No disponible     No disponible  No disponible  ...   
30048                Meta             Llano       Colombia  ...   
30049       Bogotá, D. C.  Distrito Capital       Colombia  ...   
30050       No disponible     No disponible  No disponible  ...   

       ID_AREA_CON_GR                             NME_AREA_ESP_GR  \
30046              5C                                 No registra   
30047              5B                                 No registra   
30048            3B12  Radiología, Medicina Nuclear y de Imágenes   
30049            2K02             Otras Ingenierías y Tecnologías   
30050            3B06                                 Hematología   

                           NME_AREA_GR                NME_GRAN_AREA_GR  \
30046         Ciencias de la Educación               Ciencias Sociales   
30047              Economía y Negocios               Ciencias Sociales   
30048                 Medicina Clínica  Ciencias Médicas y de la Salud   
30049  Otras Ingenierías y Tecnologías         Ingeniería y Tecnología   
30050                 Medicina Clínica  Ciencias Médicas y de la Salud   

      NME_CLASIFICACION_GR ORDEN_CLAS_GR  EDAD_ANOS_GR  \
30046                    C             2          4.91   
30047                    C             2          1.91   
30048           Reconocido             0          1.00   
30049           Reconocido             0          1.91   
30050                    C             2          1.91   

                                               INST_AVAL  \
30046               IEM ESCUELA NORMAL SUPERIOR DE PASTO   
30047                              CORPORACION INNPRENDE   
30048                                    ANGIORIENTE SAS   
30049  HPO GLOBAL RESOURCES VENTURES INC SUCURSAL COL...   
30050  PROGRAMA PARA EL TRATAMIENTO Y ESTUDIO DE ENFE...   

                                       NME_PROG_COLC1_GR  \
30046  Ciencia, Tecnología e Innovación en Ciencias H...   
30047                                          No Aplica   
30048          Ciencia, Tecnología e Innovación en Salud   
30049               Investigaciones en Energía y Minería   
30050          Ciencia, Tecnología e Innovación en Salud   

                                     NME_PROG_COLC2_GR  
30046                                        No Aplica  
30047                                        No Aplica  
30048                                    Biotecnología  
30049  Ciencia, Tecnología e Innovación en Geociencias  
30050                                    Biotecnología  

[5 rows x 21 columns]

Tanto con el head como con el tail, se pueden observar tanto la muestra inicial como la final del conjunto de datos. Observándose, que aunque haya datos en las columnas, estos aparecen como No disponible o No registra en algunas columnas; teniendo en cuenta lo anterior, se procederá a revisar la información en dichas columnas.

E primer lugar, se procederá a revisar la cantidad de datos únicos que cuenta dicha columna y con base en ello, en segundo lugar se procederá a revisar el conteo de los registros de dicha columna. Esto se realizará para las columnas mencionadas NME_AREA_ESP_GR y NME_DEPARTAMENTO_GR

In [ ]:
Clasificacion_grupos ['NME_AREA_ESP_GR'].value_counts()

No registra                                                                                 23266
Educación General (Incluye Capacitación, Pedagogía)                                           464
Negocios y Management                                                                         427
Otras Ingenierías y Tecnologías                                                               301
Ciencias Sociales, Interdisciplinaria                                                         297
                                                                                            ...  
Ingeniería del Audio                                                                            1
Planificación del Transporte y Aspectos Sociales del Transporte                                 1
Teoría Organizacional                                                                           1
Tecnología Médica de Laboratorio (Análisis de Muestras, Tecnologías para el Diagnóstico)        1
Textiles            

In [ ]:
Clasificacion_grupos['NME_DEPARTAMENTO_GR'].nunique()

33

In [ ]:
Clasificacion_grupos['NME_DEPARTAMENTO_GR'].unique()

array(['Bogotá, D. C.', 'Boyacá', 'Santander', 'Córdoba', 'Antioquia',
       'Valle del Cauca', 'La Guajira', 'Atlántico', 'Cauca',
       'Cundinamarca', 'Risaralda', 'Nariño', 'Magdalena', 'Caquetá',
       'Caldas', 'Huila', 'Bolívar', 'Quindío', 'Tolima',
       'Norte de Santander', 'Sucre', 'Meta', 'Cesar', 'Amazonas',
       'Guaviare', 'Chocó',
       'Archipiélago de San Andrés, Providencia y Santa Catalina',
       'Casanare', 'Arauca', 'Vaupés', 'No disponible', 'Vichada',
       'Putumayo'], dtype=object)

In [ ]:
Clasificacion_grupos ['NME_DEPARTAMENTO_GR'].value_counts()

Bogotá, D. C.                                               10369
Antioquia                                                    4433
Valle del Cauca                                              2427
Santander                                                    1501
Atlántico                                                    1432
Bolívar                                                      1042
Boyacá                                                        950
Caldas                                                        871
Risaralda                                                     796
Cundinamarca                                                  644
Norte de Santander                                            633
Nariño                                                        590
Cauca                                                         532
Tolima                                                        486
Quindío                                                       410
Magdalena 

Posterior, a la revisión inicial de la información se procederá a verificar los valores NaN del conjunto de datos

In [ ]:
Clasificacion_grupos.isna().sum()

ID_CONVOCATORIA           0
NME_CONVOCATORIA          0
ANO_CONVO                 0
COD_GRUPO_GR              0
NME_GRUPO_GR              4
FCREACION_GR              0
NME_MUNICIPIO_GR          0
NME_DEPARTAMENTO_GR       0
NME_REGION_GR             0
NME_PAIS_GR               0
COD_DANE_GR             270
ID_AREA_CON_GR            0
NME_AREA_ESP_GR           0
NME_AREA_GR               0
NME_GRAN_AREA_GR          0
NME_CLASIFICACION_GR      0
ORDEN_CLAS_GR             0
EDAD_ANOS_GR              0
INST_AVAL                 0
NME_PROG_COLC1_GR         0
NME_PROG_COLC2_GR         0
dtype: int64

Con base en lo anterior, se puede identificar en este Dataframe que las columnas FCREACION_GR y COD_DANE_GR son las únicas que contienen datos nulos. A continuación, se procederá a verificar el tipo de dato de las respectivas columnas

In [ ]:
Clasificacion_grupos.dtypes

ID_CONVOCATORIA           int64
NME_CONVOCATORIA         object
ANO_CONVO                object
COD_GRUPO_GR             object
NME_GRUPO_GR             object
FCREACION_GR             object
NME_MUNICIPIO_GR         object
NME_DEPARTAMENTO_GR      object
NME_REGION_GR            object
NME_PAIS_GR              object
COD_DANE_GR             float64
ID_AREA_CON_GR           object
NME_AREA_ESP_GR          object
NME_AREA_GR              object
NME_GRAN_AREA_GR         object
NME_CLASIFICACION_GR     object
ORDEN_CLAS_GR             int64
EDAD_ANOS_GR            float64
INST_AVAL                object
NME_PROG_COLC1_GR        object
NME_PROG_COLC2_GR        object
dtype: object

In [ ]:
Clasificacion_grupos['ID_CONVOCATORIA'].astype(float)

0        16.0
1        16.0
2        16.0
3        16.0
4        16.0
         ... 
30046    21.0
30047    21.0
30048    21.0
30049    21.0
30050    21.0
Name: ID_CONVOCATORIA, Length: 30051, dtype: float64

Se aprecia que la mayoría de las columnas son de tipo object. Aunque teniendo en cuenta la naturaleza del Dataframe, esta es acorde a lo requerido por el momento.

En segundo lugar, se procederá a cargar la información correspondiente a la productividad de los grupos de investigación, según las convocatorias de Minciencas para el período 2013-2021

In [7]:
Produccion_grupos_Inv=pd.read_csv('/content/drive/MyDrive/EDA_Trabajo_Final/Produccion_Grupos_Investigacion.csv')
Produccion_grupos_Inv. shape

(3166629, 14)

La información consignada en el dataframe de clasificación de grupos cuenta con 14 columnas y 3166629 filas. Sin embargo, cuando se pretende acceder a la información del conjunto de datos desde la url, se aprecia que la información, no viene completa, solo trae una parte de la información

In [ ]:
produccion=pd.read_csv('https://www.datos.gov.co/resource/33dq-ab5a.csv', sep=',')
produccion.shape

(1000, 14)

In [ ]:
Produccion_grupos_Inv.head()

ID_CONVOCATORIA          NME_CONVOCATORIA   ANO_CONVO       ID_PRODUCTO_PD  \
0             16.0  Convocatoria 640 de 2013  10/31/2013  AP-00000000000001-3   
1             16.0  Convocatoria 640 de 2013  10/31/2013   ART-0000045012-144   
2             16.0  Convocatoria 640 de 2013  10/31/2013   ART-0000045012-148   
3             16.0  Convocatoria 640 de 2013  10/31/2013   ART-0000045012-184   
4             16.0  Convocatoria 640 de 2013  10/31/2013    ART-0000045012-85   

                  NME_CLASE_PD                 NME_TIPO_MEDICION_PD  \
0  Formación de recurso humano  Formación de recursos humano Tipo B   
1           Nuevo conocimiento            Nuevo conocimiento Tipo B   
2           Nuevo conocimiento            Nuevo conocimiento Tipo B   
3           Nuevo conocimiento            Nuevo conocimiento Tipo B   
4           Nuevo conocimiento            Nuevo conocimiento Tipo B   

             NME_TIPOLOGIA_PD ID_TIPO_PD_MED  \
0          Apoyo de programas        AP-AP_B   
1  Artículos de investigación      ART-ART_D   
2  Artículos de investigación      ART-ART_D   
3  Artículos de investigación      ART-ART_D   
4  Artículos de investigación      ART-ART_D   

                           NME_CATEGORIA_PD FCREACION_PD  \
0          Apoyo de programas Con Calidad B   01/10/2011   
1  Artículos de investigación Con Calidad D   01/10/2010   
2  Artículos de investigación Con Calidad D   01/10/2012   
3  Artículos de investigación Con Calidad D   01/04/2013   
4  Artículos de investigación Con Calidad D   01/04/2009   

                                     NME_PRODUCTO_PD COD_GRUPO_GR  \
0  Maestría en Investigación en problemas Sociale...   COL0016283   
1                            La moral de la crueldad   COL0016283   
2               La ilusión llamada páramo de Rabanal   COL0016283   
3            A la memoria de Guillermo Hoyos Vásquez   COL0016283   
4  Aspectos del devenir militante del revoluciona...   COL0016283   

                NME_GRUPO_GR  ID_PERSONA_PD  
0  Socialización y violencia              0  
1  Socialización y violencia          45012  
2  Socialización y violencia          45012  
3  Socialización y violencia          45012  
4  Socialización y violencia          45012

In [ ]:
Produccion_grupos_Inv.tail()

ID_CONVOCATORIA          NME_CONVOCATORIA   ANO_CONVO  \
3166624             21.0  Convocatoria 894 de 2021  02/25/2021   
3166625             21.0  Convocatoria 894 de 2021  02/25/2021   
3166626             21.0  Convocatoria 894 de 2021  02/25/2021   
3166627             21.0  Convocatoria 894 de 2021  02/25/2021   
3166628             21.0  Convocatoria 894 de 2021  02/25/2021   

           ID_PRODUCTO_PD                                       NME_CLASE_PD  \
3166624  ART-0000919497-4                                 Nuevo conocimiento   
3166625  ART-0000919497-5                                 Nuevo conocimiento   
3166626  IPP-0000919497-7                Desarrollo tecnológico e innovación   
3166627  PID-0000919497-1                        Formación de recurso humano   
3166628   RC-0001425704-1  Apropiación social del conocimiento y divulgac...   

                                      NME_TIPO_MEDICION_PD  \
3166624                          Nuevo conocimiento Tipo A   
3166625                             Nuevo conocimiento Top   
3166626                          Nuevo conocimiento Tipo B   
3166627                Formación de recursos humano Tipo B   
3166628  Apropiación social del conocimiento y divulgac...   

                                 NME_TIPOLOGIA_PD ID_TIPO_PD_MED  \
3166624                Artículos de investigación      ART-ART_B   
3166625                Artículos de investigación     ART-ART_A1   
3166626  Innovaciones en procedimiento y servicio        IPP-IPP   
3166627    Proyecto de Investigacion y Desarrollo      PID-PID_C   
3166628         Red de conocimiento especializado        RC-RC_A   

                                          NME_CATEGORIA_PD FCREACION_PD  \
3166624           Artículos de investigación Con Calidad B   01/01/2020   
3166625          Artículos de investigación Con Calidad A1   01/02/2019   
3166626  Innovaciones en procedimiento y servicio Con C...   01/07/2020   
3166627  Proyecto de Investigacion y Desarrollo Con Cal...   01/08/2019   
3166628    Red de conocimiento especializado Con Calidad A   01/02/2019   

                                           NME_PRODUCTO_PD COD_GRUPO_GR  \
3166624  Reactivation of chagas disease after autologou...   COL0224713   
3166625  Autologous hematopoietic stem cell transplanta...   COL0224713   
3166626  Innovación pedagógica en el proceso de capacit...   COL0224713   
3166627  Registro institucional de trasplante de progen...   COL0224713   
3166628  PETHEMA - Programa Español de Tratamientos en ...   COL0224713   

                    NME_GRUPO_GR  ID_PERSONA_PD  
3166624  Hematología y Oncología         919497  
3166625  Hematología y Oncología         919497  
3166626  Hematología y Oncología         919497  
3166627  Hematología y Oncología         919497  
3166628  Hematología y Oncología        1425704

Tanto con el head como con el tail, se pueden observar tanto la muestra inicial como la final del conjunto de datos. Observándose, que  para este conjunto si aparecen NaN

In [ ]:
Produccion_grupos_Inv.isna().sum()

ID_CONVOCATORIA            0
NME_CONVOCATORIA           0
ANO_CONVO                  0
ID_PRODUCTO_PD             0
NME_CLASE_PD               0
NME_TIPO_MEDICION_PD       0
NME_TIPOLOGIA_PD           0
ID_TIPO_PD_MED             0
NME_CATEGORIA_PD           0
FCREACION_PD               0
NME_PRODUCTO_PD         2843
COD_GRUPO_GR               0
NME_GRUPO_GR              92
ID_PERSONA_PD              0
dtype: int64

Con base en lo anterior, se puede identificar en este Dataframe que las columnas NME_PRODUCTO_PD y NME_GRUPO_GR son las únicas que contienen datos nulos. A continuación, se procederá a verificar el tipo de dato de las respectivas columnas

In [ ]:
Produccion_grupos_Inv.dtypes

ID_CONVOCATORIA         float64
NME_CONVOCATORIA         object
ANO_CONVO                object
ID_PRODUCTO_PD           object
NME_CLASE_PD             object
NME_TIPO_MEDICION_PD     object
NME_TIPOLOGIA_PD         object
ID_TIPO_PD_MED           object
NME_CATEGORIA_PD         object
FCREACION_PD             object
NME_PRODUCTO_PD          object
COD_GRUPO_GR             object
NME_GRUPO_GR             object
ID_PERSONA_PD             int64
dtype: object

In [ ]:
Produccion_grupos_Inv.dtypes

ID_CONVOCATORIA         float64
NME_CONVOCATORIA         object
ANO_CONVO                object
ID_PRODUCTO_PD           object
NME_CLASE_PD             object
NME_TIPO_MEDICION_PD     object
NME_TIPOLOGIA_PD         object
ID_TIPO_PD_MED           object
NME_CATEGORIA_PD         object
FCREACION_PD             object
NME_PRODUCTO_PD          object
COD_GRUPO_GR             object
NME_GRUPO_GR             object
ID_PERSONA_PD             int64
dtype: object

Al igual que en la daset anterior, se aprecia que la mayoría de las columnas son de tipo object. Aunque teniendo en cuenta la naturaleza de los datos, esta es acorde a lo requerido por el momento.

Con el propósito de unificar la información relacionada a productividad de grupos de investigación y la de clasificación de los grupos y dar respuesta tanto a la pregunta como al objetivo. Se procederá a realizar una unión entre ambos Dataframes a través de las columnas  **ID_CONVOCATORIA** y **COD_GRUPO_GR**

In [ ]:
Info=pd.merge(Clasificacion_grupos,Produccion_grupos_Inv, how='inner', on=['COD_GRUPO_GR'])
Info.shape

MemoryError: Unable to allocate 290. MiB for an array with shape (12, 3166629) and data type object

La información consignada en el dataframe unificado (clasificación de grupos y productividad de los grupos) cuenta con 33 columnas y 3166629 filas. A continuación, se procedera a visualizar el head y tail del conjunto de datos unificados.

In [ ]:
Info.head()

key_0  ID_CONVOCATORIA_x        NME_CONVOCATORIA_x ANO_CONVO_x  \
0  COL0016283                 16  Convocatoria 640 de 2013  31/10/2013   
1  COL0016283                 16  Convocatoria 640 de 2013  31/10/2013   
2  COL0016283                 16  Convocatoria 640 de 2013  31/10/2013   
3  COL0016283                 16  Convocatoria 640 de 2013  31/10/2013   
4  COL0016283                 16  Convocatoria 640 de 2013  31/10/2013   

  COD_GRUPO_GR_x             NME_GRUPO_GR_x FCREACION_GR NME_MUNICIPIO_GR  \
0     COL0016283  Socialización y violencia   01/02/1993     Bogotá, D.C.   
1     COL0016283  Socialización y violencia   01/02/1993     Bogotá, D.C.   
2     COL0016283  Socialización y violencia   01/02/1993     Bogotá, D.C.   
3     COL0016283  Socialización y violencia   01/02/1993     Bogotá, D.C.   
4     COL0016283  Socialización y violencia   01/02/1993     Bogotá, D.C.   

  NME_DEPARTAMENTO_GR     NME_REGION_GR  ...                 NME_CLASE_PD  \
0       Bogotá, D. C.  Distrito Capital  ...  Formación de recurso humano   
1       Bogotá, D. C.  Distrito Capital  ...           Nuevo conocimiento   
2       Bogotá, D. C.  Distrito Capital  ...           Nuevo conocimiento   
3       Bogotá, D. C.  Distrito Capital  ...           Nuevo conocimiento   
4       Bogotá, D. C.  Distrito Capital  ...           Nuevo conocimiento   

                  NME_TIPO_MEDICION_PD            NME_TIPOLOGIA_PD  \
0  Formación de recursos humano Tipo B          Apoyo de programas   
1            Nuevo conocimiento Tipo B  Artículos de investigación   
2            Nuevo conocimiento Tipo B  Artículos de investigación   
3            Nuevo conocimiento Tipo B  Artículos de investigación   
4            Nuevo conocimiento Tipo B  Artículos de investigación   

  ID_TIPO_PD_MED                          NME_CATEGORIA_PD FCREACION_PD  \
0        AP-AP_B          Apoyo de programas Con Calidad B   01/10/2011   
1      ART-ART_D  Artículos de investigación Con Calidad D   01/10/2010   
2      ART-ART_D  Artículos de investigación Con Calidad D   01/10/2012   
3      ART-ART_D  Artículos de investigación Con Calidad D   01/04/2013   
4      ART-ART_D  Artículos de investigación Con Calidad D   01/04/2009   

                                     NME_PRODUCTO_PD  COD_GRUPO_GR_y  \
0  Maestría en Investigación en problemas Sociale...      COL0016283   
1                            La moral de la crueldad      COL0016283   
2               La ilusión llamada páramo de Rabanal      COL0016283   
3            A la memoria de Guillermo Hoyos Vásquez      COL0016283   
4  Aspectos del devenir militante del revoluciona...      COL0016283   

              NME_GRUPO_GR_y ID_PERSONA_PD  
0  Socialización y violencia           0.0  
1  Socialización y violencia       45012.0  
2  Socialización y violencia       45012.0  
3  Socialización y violencia       45012.0  
4  Socialización y violencia       45012.0  

[5 rows x 36 columns]

In [ ]:
Info.tail()

key_0  ID_CONVOCATORIA_x        NME_CONVOCATORIA_x ANO_CONVO_x  \
16829717  COL0224713                 21  Convocatoria 894 de 2021  25/02/2021   
16829718  COL0224713                 21  Convocatoria 894 de 2021  25/02/2021   
16829719  COL0224713                 21  Convocatoria 894 de 2021  25/02/2021   
16829720  COL0224713                 21  Convocatoria 894 de 2021  25/02/2021   
16829721  COL0224713                 21  Convocatoria 894 de 2021  25/02/2021   

         COD_GRUPO_GR_x           NME_GRUPO_GR_x FCREACION_GR  \
16829717     COL0224713  Hematología y Oncología   01/01/2019   
16829718     COL0224713  Hematología y Oncología   01/01/2019   
16829719     COL0224713  Hematología y Oncología   01/01/2019   
16829720     COL0224713  Hematología y Oncología   01/01/2019   
16829721     COL0224713  Hematología y Oncología   01/01/2019   

         NME_MUNICIPIO_GR NME_DEPARTAMENTO_GR  NME_REGION_GR  ...  \
16829717    No disponible       No disponible  No disponible  ...   
16829718    No disponible       No disponible  No disponible  ...   
16829719    No disponible       No disponible  No disponible  ...   
16829720    No disponible       No disponible  No disponible  ...   
16829721    No disponible       No disponible  No disponible  ...   

                                               NME_CLASE_PD  \
16829717                                 Nuevo conocimiento   
16829718                                 Nuevo conocimiento   
16829719                Desarrollo tecnológico e innovación   
16829720                        Formación de recurso humano   
16829721  Apropiación social del conocimiento y divulgac...   

                                       NME_TIPO_MEDICION_PD  \
16829717                          Nuevo conocimiento Tipo A   
16829718                             Nuevo conocimiento Top   
16829719                          Nuevo conocimiento Tipo B   
16829720                Formación de recursos humano Tipo B   
16829721  Apropiación social del conocimiento y divulgac...   

                                  NME_TIPOLOGIA_PD ID_TIPO_PD_MED  \
16829717                Artículos de investigación      ART-ART_B   
16829718                Artículos de investigación     ART-ART_A1   
16829719  Innovaciones en procedimiento y servicio        IPP-IPP   
16829720    Proyecto de Investigacion y Desarrollo      PID-PID_C   
16829721         Red de conocimiento especializado        RC-RC_A   

                                           NME_CATEGORIA_PD FCREACION_PD  \
16829717           Artículos de investigación Con Calidad B   01/01/2020   
16829718          Artículos de investigación Con Calidad A1   01/02/2019   
16829719  Innovaciones en procedimiento y servicio Con C...   01/07/2020   
16829720  Proyecto de Investigacion y Desarrollo Con Cal...   01/08/2019   
16829721    Red de conocimiento especializado Con Calidad A   01/02/2019   

                                            NME_PRODUCTO_PD  COD_GRUPO_GR_y  \
16829717  Reactivation of chagas disease after autologou...      COL0224713   
16829718  Autologous hematopoietic stem cell transplanta...      COL0224713   
16829719  Innovación pedagógica en el proceso de capacit...      COL0224713   
16829720  Registro institucional de trasplante de progen...      COL0224713   
16829721  PETHEMA - Programa Español de Tratamientos en ...      COL0224713   

                   NME_GRUPO_GR_y ID_PERSONA_PD  
16829717  Hematología y Oncología      919497.0  
16829718  Hematología y Oncología      919497.0  
16829719  Hematología y Oncología      919497.0  
16829720  Hematología y Oncología      919497.0  
16829721  Hematología y Oncología     1425704.0  

[5 rows x 36 columns]

In [ ]:
Info.isna().sum()

key_0                       0
ID_CONVOCATORIA_x           0
NME_CONVOCATORIA_x          0
ANO_CONVO_x                 0
COD_GRUPO_GR_x              0
NME_GRUPO_GR_x             92
FCREACION_GR                0
NME_MUNICIPIO_GR            0
NME_DEPARTAMENTO_GR         0
NME_REGION_GR               0
NME_PAIS_GR                 0
COD_DANE_GR             27586
ID_AREA_CON_GR              0
NME_AREA_ESP_GR             0
NME_AREA_GR                 0
NME_GRAN_AREA_GR            0
NME_CLASIFICACION_GR        0
ORDEN_CLAS_GR               0
EDAD_ANOS_GR                0
INST_AVAL                   0
NME_PROG_COLC1_GR           0
NME_PROG_COLC2_GR           0
ID_CONVOCATORIA_y           3
NME_CONVOCATORIA_y          3
ANO_CONVO_y                 3
ID_PRODUCTO_PD              3
NME_CLASE_PD                3
NME_TIPO_MEDICION_PD        3
NME_TIPOLOGIA_PD            3
ID_TIPO_PD_MED              3
NME_CATEGORIA_PD            3
FCREACION_PD                3
NME_PRODUCTO_PD         15645
COD_GRUPO_

Teniendo en cuenta el proceso de unión de ambos conjuntos de datos, se puede concluir que la información faltante de ambos conjuntos de datos se mantiene. En este sentido, se puede concluir que los elementos relacionados a 'NME_AREA_ESP_GR' y 'NME_DEPARTAMENTO_GR' se mantienen con sus respectivos valores de "No registra" o "No disponible"

In [ ]:
Info.dtypes

key_0                    object
ID_CONVOCATORIA_x         int64
NME_CONVOCATORIA_x       object
ANO_CONVO_x              object
COD_GRUPO_GR_x           object
NME_GRUPO_GR_x           object
FCREACION_GR             object
NME_MUNICIPIO_GR         object
NME_DEPARTAMENTO_GR      object
NME_REGION_GR            object
NME_PAIS_GR              object
COD_DANE_GR             float64
ID_AREA_CON_GR           object
NME_AREA_ESP_GR          object
NME_AREA_GR              object
NME_GRAN_AREA_GR         object
NME_CLASIFICACION_GR     object
ORDEN_CLAS_GR             int64
EDAD_ANOS_GR            float64
INST_AVAL                object
NME_PROG_COLC1_GR        object
NME_PROG_COLC2_GR        object
ID_CONVOCATORIA_y       float64
NME_CONVOCATORIA_y       object
ANO_CONVO_y              object
ID_PRODUCTO_PD           object
NME_CLASE_PD             object
NME_TIPO_MEDICION_PD     object
NME_TIPOLOGIA_PD         object
ID_TIPO_PD_MED           object
NME_CATEGORIA_PD         object
FCREACIO

**5.2) Definición de variables a trabajar**

Teniendo en cuenta que la unión de los dos (2) conjuntos de datos, traían información similar se procederá a suprimir dicha información y así construir un conjunto de datos con el menor número de variables pero que sean necesarias para dar respuesta al objetivo planteado. Para lo cual, teniendo en cuenta el modelo de medición, actual hasta el momento, se dejarán las siguientes variables:
             
ANO_CONVO_x

COD_GRUPO_GR

EDAD_ANOS_GR

NME_GRAN_AREA_GR

NME_CLASIFICACION_GR

ORDEN_CLAS_GR

NME_CLASE_PD

NME_TIPO_MEDICION_PD

NME_TIPOLOGIA_PD

NME_CATEGORIA_PD

FCREACION_PD

NME_DEPARTAMENTO_GR

In [ ]:
df_final=Info[['ANO_CONVO_x', 'COD_GRUPO_GR_x', 'EDAD_ANOS_GR', 'NME_GRAN_AREA_GR', 'NME_CLASIFICACION_GR','ORDEN_CLAS_GR',
              'NME_CLASE_PD', 'NME_TIPO_MEDICION_PD', 'NME_TIPOLOGIA_PD', 'NME_CATEGORIA_PD', 'FCREACION_PD', 'NME_DEPARTAMENTO_GR']]
df_final.shape

(16829722, 12)

In [ ]:
df_final.isna().sum()

ANO_CONVO_x             0
COD_GRUPO_GR            0
EDAD_ANOS_GR            0
NME_GRAN_AREA_GR        0
NME_CLASIFICACION_GR    0
ORDEN_CLAS_GR           0
NME_CLASE_PD            0
NME_TIPO_MEDICION_PD    0
NME_TIPOLOGIA_PD        0
NME_CATEGORIA_PD        0
FCREACION_PD            0
NME_DEPARTAMENTO_GR     0
dtype: int64

In [ ]:
df_final.dtypes

ANO_CONVO_x              object
COD_GRUPO_GR             object
EDAD_ANOS_GR            float64
NME_GRAN_AREA_GR         object
NME_CLASIFICACION_GR     object
ORDEN_CLAS_GR             int64
NME_CLASE_PD             object
NME_TIPO_MEDICION_PD     object
NME_TIPOLOGIA_PD         object
NME_CATEGORIA_PD         object
FCREACION_PD             object
NME_DEPARTAMENTO_GR      object
dtype: object

Ahora se procederá a modificar el tipo de datos de las columnas ANO_CONVO_x y FCREACION_PD a tipo fecha; aunque lo que interesa de esta es el año.

In [ ]:
df_final['ANO_CONVO_x']=pd.to_datetime(df_final['ANO_CONVO_x'])

C:\Users\Jovany\AppData\Local\Temp\ipykernel_11548\1191217539.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_final['ANO_CONVO_x']=pd.to_datetime(df_final['ANO_CONVO_x'])


In [ ]:
df_final ['Anio_convocatoria']=df_final['ANO_CONVO_x'].dt.year

C:\Users\Jovany\AppData\Local\Temp\ipykernel_11548\3173904744.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_final ['Anio_convocatoria']=df_final['ANO_CONVO_x'].dt.year


In [ ]:
#Se agrega el parametro errors='coerce' debido a que sin este parametro aparecía un error para realizar la modificación
df_final['FCREACION_PD']=pd.to_datetime(df_final['FCREACION_PD'], errors = 'coerce')

C:\Users\Jovany\AppData\Local\Temp\ipykernel_11548\2697167400.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_final['FCREACION_PD']=pd.to_datetime(df_final['FCREACION_PD'], errors = 'coerce')


In [ ]:
df_final.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16829719 entries, 0 to 16829718
Data columns (total 13 columns):
 #   Column                Dtype         
---  ------                -----         
 0   ANO_CONVO_x           datetime64[ns]
 1   COD_GRUPO_GR          object        
 2   EDAD_ANOS_GR          float64       
 3   NME_GRAN_AREA_GR      object        
 4   NME_CLASIFICACION_GR  object        
 5   ORDEN_CLAS_GR         int64         
 6   NME_CLASE_PD          object        
 7   NME_TIPO_MEDICION_PD  object        
 8   NME_TIPOLOGIA_PD      object        
 9   NME_CATEGORIA_PD      object        
 10  FCREACION_PD          datetime64[ns]
 11  NME_DEPARTAMENTO_GR   object        
 12  Anio_convocatoria     int64         
dtypes: datetime64[ns](2), float64(1), int64(2), object(8)
memory usage: 1.8+ GB


**5.3)**Análisis de variables**

De acuerdo a las variables seleccionadas, se procederá a realizar un análisis teniendo en cuenta la clasificación de los grupos a través del tiempo; las principales áreas en las que éstos se encuentran adheridos, así como la clasificación de los productos elaborados por cada uno de los diferentes grupos. Y por último, identificar los departamentos donde se ubican los grupos de investigación y así identificar las zonas geográficas donde se concentra el proceso de investigación anivel nacional.

In [ ]:
clasificacion=df_final.groupby(['Anio_convocatoria','NME_CLASIFICACION_GR', 'COD_GRUPO_GR']).value_counts()['COD_GRUPO_GR']
clasificacion

#clasificacion=df_final.pivot_table(index=['Anio_convocatoria','NME_CLASIFICACION_GR'],
                                 # columns=['NME_CLASIFICACION_GR'], aggfunc='value_counts', fill_value=0)
#clasificacion

KeyError: 'COD_GRUPO_GR'

**Se realizará una tabla pivote**

In [ ]:
Dinamica=Info.pivot_table(index=['NME_CONVOCATORIA_x', 'COD_GRUPO_GR', 'NME_CLASIFICACION_GR'],
                          columns=['NME_CLASE_PD'], aggfunc='count', fill_value=0)
Dinamica

ANO_CONVO_x  \
NME_CLASE_PD                                               Apropiación social del conocimiento   
NME_CONVOCATORIA_x       COD_GRUPO_GR NME_CLASIFICACION_GR                                       
Convocatoria 640 de 2013 COL0000013   C                                                      2   
                         COL0000049   B                                                     17   
                         COL0000059   B                                                      8   
                         COL0000069   D                                                      2   
                         COL0000078   B                                                      1   
...                                                                                        ...   
Convocatoria 894 de 2021 COL0224491   C                                                      0   
                         COL0224571   C                                                      0   
                         COL0224615   Reconocido                                             0   
                         COL0224642   Reconocido                                             0   
                         COL0224713   C                                                      0   

                                                                                                                                    \
NME_CLASE_PD                                               Apropiación social del conocimiento y divulgación pública de la ciencia   
NME_CONVOCATORIA_x       COD_GRUPO_GR NME_CLASIFICACION_GR                                                                           
Convocatoria 640 de 2013 COL0000013   C                                                                     0                        
                         COL0000049   B                                                                     0                        
                         COL0000059   B                                                                     0                        
                         COL0000069   D                                                                     0                        
                         COL0000078   B                                                                     0                        
...                                                                                                       ...                        
Convocatoria 894 de 2021 COL0224491   C                                                                    18                        
                         COL0224571   C                                                                     3                        
                         COL0224615   Reconocido                                                            0                        
                         COL0224642   Reconocido                                                            4                        
                         COL0224713   C                                                                     1                        

                                                                                                \
NME_CLASE_PD                                               Desarrollo tecnológico e innovación   
NME_CONVOCATORIA_x       COD_GRUPO_GR NME_CLASIFICACION_GR                                       
Convocatoria 640 de 2013 COL0000013   C                                                      0   
                         COL0000049   B                                                      0   
                         COL0000059   B                                                      0   
                         COL0000069   D                                                      0   
                         COL0000078   B                                                      0   
...                                              

In [ ]:
Dinamica.describe()

ANO_CONVO_x  \
NME_CLASE_PD Apropiación social del conocimiento   
count                               30046.000000   
mean                                   29.833256   
std                                    46.599610   
min                                     0.000000   
25%                                     1.000000   
50%                                    15.000000   
75%                                    40.000000   
max                                  1370.000000   

                                                                                      \
NME_CLASE_PD Apropiación social del conocimiento y divulgación pública de la ciencia   
count                                              30046.000000                        
mean                                                  11.025128                        
std                                                   37.576408                        
min                                                    0.000000                        
25%                                                    0.000000                        
50%                                                    0.000000                        
75%                                                    0.000000                        
max                                                 1344.000000                        

                                                                              \
NME_CLASE_PD Desarrollo tecnológico e innovación Formación de recurso humano   
count                               30046.000000                30046.000000   
mean                                    3.124542                   36.464821   
std                                    13.415254                   47.039684   
min                                     0.000000                    0.000000   
25%                                     0.000000                    9.000000   
50%                                     0.000000                   22.000000   
75%                                     1.000000                   46.000000   
max                                   556.000000                  875.000000   

                                                 \
NME_CLASE_PD No categorizado Nuevo conocimiento   
count           30046.000000       30046.000000   
mean                0.008920          24.936031   
std                 0.212099          33.076395   
min                 0.000000           0.000000   
25%                 0.000000           7.000000   
50%                 0.000000          15.000000   
75%                 0.000000          31.000000   
max                17.000000        1070.000000   

                                     ANO_CONVO_y  \
NME_CLASE_PD Apropiación social del conocimiento   
count                               30046.000000   
mean                                   29.833256   
std                                    46.599610   
min                                     0.000000   
25%                                     1.000000   
50%                                    15.000000   
75%                                    40.000000   
max                                  1370.000000   

                                                                                      \
NME_CLASE_PD Apropiación social del conocimiento y divulgación pública de la ciencia   
count                                              30046.000000                        
mean                                                  11.025128                        
std                                                   37.576408                        
min                                                    0.000000                        
25%                                                    0.000000                        
50%                                                    0.000000                        
75%                                                    0.000000                        
max       